src/main/java/com/snowflake/kafka/connector/internal/streaming/StreamingClientProvider.java

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (c) 2023 Snowflake Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.snowflake.kafka.connector.internal.streaming;
+
+import static com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig.SNOWPIPE_STREAMING_FILE_VERSION;
+import static net.snowflake.ingest.utils.ParameterProvider.BLOB_FORMAT_VERSION;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.snowflake.kafka.connector.SnowflakeSinkConnectorConfig;
+import com.snowflake.kafka.connector.Utils;
+import com.snowflake.kafka.connector.internal.KCLogger;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Properties;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClient;
+import net.snowflake.ingest.streaming.SnowflakeStreamingIngestClientFactory;
+import net.snowflake.ingest.utils.SFException;
+import org.apache.kafka.connect.errors.ConnectException;
+
+/** Singleton that provides the streaming client */
+public class StreamingClientProvider {
+  public static final StreamingClientProvider streamingClientProvider =
+      new StreamingClientProvider();
```

Comment:
I am personally not a fan of public static instance. 
I would rather make this a private and have a getter which returns the singleton `StreamingClientProvider`. This in true sense will make this a singleton. 


```suggestion
private static class Holder {
    private static final StreamingClientProvider INSTANCE = new StreamingClientProvider();
  }

  public static StreamingClientProvider getStreamingClientProviderInstance() {
    return Holder.INSTANCE;
  }

```

Also add docstring to public method.

Reply:
I changed it to a public static method that will return the instance that is stored inside a private static subclass